lib/neo4j/active_node/query/query_proxy_enumerable.rb

```diff
@@ -14,6 +14,8 @@ def each(node = true, rel = nil, &block)
         end
 
         def result(node = true, rel = nil)
+          return [].freeze if @start_object && @start_object.new_record?
```

Comment:
Indeed, I think it prevents querying when accessing directly to associations.

I'll probably need to write another spec and a fix for your case. Good catch, thanks!